FinderCompass/src/main/java/atomicstryker/findercompass/client/CompassSetting.java

```diff
@@ -56,6 +57,21 @@ public boolean isStrongholdNeedleEnabled()
         return customNeedles;
     }
     
+    public CompassTargetData getCustomNeedle(String pOreDictName)
+    {
+    	CompassTargetData needle = null;
+    	if (pOreDictName != null){
+    		for (CompassTargetData entry : customNeedles.keySet()) {
+				if (entry.getOreDictName().equals(pOreDictName))

```

Comment:
Is this never null? Otherwise flipping entry.getOreDictName() and pOreDictName prevents theoretical NPE